answers/w11.md

```diff
@@ -0,0 +1,183 @@
+### What are constructor functions? How do we use them?
+
+Constructor functions define a new instance of a Javascript object. We create a Javascript object using the `new` keyword.  
+
+```js
+var Planet = function( info ){
+  this.name = info.name;
+  this.distance = info.distance;
+}
+
+var neptune = new Planet({ name: 'Neptune', distance: 'far away' });
+```
+
+### What is a prototype? Why do we care about them?
+
+Prototypes define the default -- or "blueprint" -- values and methods of an object.  
+
+### What action do you expect this route to perform: `app.post('/authors', function() {...}`?
+
+It is a `POST` request, so it will create a new `author` in our database.
+
+### Write the url to request all authors from a Rails app, formatted as JSON.
+
+`http://localhost:3000/authors.json`
+
+### Create a media query that supports most mobile phones.
+
+```css
+@media only screen and (max-device-width: 480px) {
+  /* ... */
+}
+```
+
+### How can box sizing affect a DOM Element?
+
+Depending on the value of a DOM element's `box-sizing` property, its width may be calculated differently.  
+* If `box-sizing: content-box`, width is equal to an element's content.
+* If `box-sizing: border-box`, width is equal to the sum of an element's content, padding and border. Not margin.
+
+### Explain the difference between referencing and invoking a function.
+
+To invoke a function is to call it. To reference a function is to return that function's definition.  
+
+```js
+// Function definition
+function sayHello(){
+  return "Hi."
+}
+
+// Invoke
+sayHello();
+// => "Hi."
+
+// Reference
+sayHello;
+/* => function sayHello(){
+  return "Hi."
+}
+*/
+```
+
+### Explain the concept of a 'callback' function.
+
+Callbacks are functions that are passed in as arguments to another function so that they can be executed later when it is called. An example of a callback is the `helloWorld` function passed into the below event listener...
+
+```js
+// Callback function definition.
+function helloWorld(){
+  console.log( "Hello world.");
+}
+
+// helloWorld is invoked whenever a button is clicked.
+$("button").on("click", helloWorld);
+```
+
+### Explain what an ORM is and why they are useful.
+
+An ORM ("Object Relational Mapping") is an interface that allows us to represent items in a database as objects on a server (and vice-versa).  
+An example of an ORM is Active Record. In AR, models and tables represent each other. Associations are handled using foreign keys.  
+
+### List at least 3 methods added to a class when it inherits from ActiveRecord::Base.
+
+`.create`  
+`.destroy`  
+`.find_by`  
+
+### Explain what REST is and why we use it.
+
+REST stands for "Representational State Transfer." To be "RESTful" is to use HTTP requests (e.g., `GET`, `POST`) to execute CRUD actions on data.  
```

Comment:
REST stands for "Representational State Transfer." To be "RESTful" is represent each resource with the same 7 actions (each a combination of HTTP method and path), to execute CRUD actions on data.